src/main/java/net/ossindex/version/VersionFactory.java

```diff
@@ -48,42 +49,61 @@
 import org.antlr.v4.runtime.CommonTokenStream;
 import org.antlr.v4.runtime.tree.ParseTreeWalker;
 
-/** This factory provides an appropriate version implementation for
+/**
+ * This factory provides an appropriate version implementation for
  * specified version strings.
  *
  * @author Ken Duck
- *
  */
 public class VersionFactory
 {
+  private static VersionFactory strictInstance;
+
   private static VersionFactory instance;
 
+  private boolean strict = false;
```

Comment:
Can you make `strict` final?  That would help me understand if this is a switch can can't be reset.

Reply:
Indeed I can (and will!)